dietstory-routing/registration/views.py

```diff
@@ -670,5 +674,100 @@ def post(self, request, *args, **kwargs):
         
         logger.warn("[RV-10] Invalid input parameters")
         return JsonResponse({'message': "Invalid input parameters"}, status=status.HTTP_400_BAD_REQUEST)
+    
+    class DisconnectFromServerView(views.APIView):
 
+        permission_classes = (permissions.IsAuthenticated,)
+
+class DisconnectView(views.APIView):
+    permission_classes = (permissions.AllowAny,)
 
+    """
+    summary: Disconnect character
+    description: Disconnect character based on their characters name
+                 and what world they are in.
+    parameters:
+        - name: character_id
+            schema:
+                type: string
+            description: >
+                The name of the character to dc.
+            required: true
+        - name: world
+            schema:
+                type: string
+            description: >
+                The world the character belongs in.
+            required: true
+    tags:
+        - DisconnectView
+    responses:
+        200:
+            content:
+                application/json:
+                    schema:
+                        type: object
+                        properties:
+                                message:
+                                    type: string
+                                    description: Successful disconnect.
+        404:
+            content:
+                application/json:
+                    schema:
+                        type: object
+                        properties:
+                                message:
+                                    type: string
+                                    description: A character with that name does not exist in
+                                                 your account.
+        400:
+            content:
+                application/json:
+                    schema:
+                        type: object
+                        properties:
+                                message:
+                                    type: string
+                                    description: There was an error disconnecting
+                                                 that character.
+    """
+    def post(self, request, *args, **kwargs):
+        logger.info("Got a request to DC a character.")
+        
+        params = DisconnectForm(request.data)
+        if params.is_valid():
+            username = request.session['username']
```

Comment:
If our session includes an account id, use that instead. Account id is a primary key as opposed to a full-table query using username.

Reply:
Looks like our session only includes the username. We could potentially store the account id instead though. But that would have to be done during Login. (https://github.com/allenaerostar/client-launcher/blob/master/dietstory-routing/registration/views.py#L449)